### PR TITLE
Fixes #2185 Follow up on dependency lib version needed

### LINKED ISSUE #2185: Getting Elastic 8.x to GA
Lots of work has been over the past year to make Elastica compatible with Elasticsearch 8.x. Also many discussions have happened. This issue is to track the missing bits to make it GA and also have a single place to link together all discussions and PRs.

## TODO

* [x] PR to get Elastica 8 to alpha: https://github.com/ruflin/Elastica/pull/2181
* [x] Follow up on dependency lib version needed: https://github.com/ruflin/Elastica/pull/2181#issuecomment-1826405139 https://github.com/ruflin/Elastica/pull/2186
* [x] Update README files with 8.x notes
* [x] Update changelog with breaking changes
* [x] Add migration guide from 7.x to 8.x https://github.com/ruflin/Elastica/pull/2214
* [x] https://github.com/ruflin/Elastica/pull/2188

## Links

* [Discussions around Elastica 8.0](https://github.com/ruflin/Elastica/discussions/1961)
* [Initial 8 PR that kicked of the work](https://github.com/ruflin/Elastica/pull/2168)

### PR DESCRIPTION
**Pull request by pawelkeska**
@sidz `ClientInterface` was added from 8.4.1 i've changed composer.json

@rufin i added some tests to connection and fix bug with path in `Client`

### CODE REVIEW

**Review by sidz — APPROVED** [review]
Brilliant. thanks @pawelkeska :+1:

File: src/Connection.php [hunk 1]
```diff
@@ -167,14 +167,6 @@ public function getTransportObject(): Transport
         return $this->getTransport();
     }
 
-    /**
```
Comment (ruflin) [inline src/Connection.php:170]: This goes away because it is not supported anymore?

**Review by ruflin — APPROVED** [review]
Change LGTM. Before we merge, quickly get some details on why some parts are removed (I guess not supported anymore)

**Comment by pawelkeska** [comment]
@ruflin yes, `isPersistent` and `getAuthType` was used by transports elastica. It's not anymore need.

**Pull request merged**